setup.py

```diff
@@ -122,41 +118,49 @@ def _set_manifest_path(manifest_dir: str, aggregate: bool = False) -> Generator:
-        pkg_setup = os.path.join(pkg_path, "__setup__.py")
-        if os.path.exists(pkg_setup):
-            print(f"{pkg_setup} exists. Running `setuptools.setup`")
-            setup_module = _load_py_module(name=f"{pkg}_setup", location=pkg_setup)
-            setup_args = setup_module._setup_args()
-            is_main_pkg = pkg == "lightning"
-            if is_wheel_install and not is_main_pkg:
-                setuptools.setup(**setup_args)
-            else:
-                # we are installing from source, set the correct manifest path
-                with _set_manifest_path(pkg_path, aggregate=is_main_pkg):
-                    setuptools.setup(**setup_args)
-            break
+    print(f"Requested package: '{_PACKAGE_NAME}'")  # requires `-v` to appear
+
+    local_pkgs = [
+        os.path.basename(p)
+        for p in glob.glob(os.path.join(_PATH_SRC, "*"))
+        if os.path.isdir(p) and not p.endswith(".egg-info")
+    ]
+    print(f"Local package candidates: {local_pkgs}")
+    is_source_install = len(local_pkgs) > 2
+    print(f"Installing from source: {is_source_install}")
+    if is_source_install:
+        if _PACKAGE_NAME is not None and _PACKAGE_NAME not in _PACKAGE_MAPPING:
+            raise ValueError(
+                f"Unexpected package name: {_PACKAGE_NAME}. Possible choices are: {list(_PACKAGE_MAPPING)}"
+            )
+        package_to_install = _PACKAGE_MAPPING.get(_PACKAGE_NAME, "lightning")
+        if package_to_install == "lightning":  # install everything
+            # merge all requirements files
+            assistant._load_aggregate_requirements(_PATH_REQUIRE, _FREEZE_REQUIREMENTS)
+            # replace imports and copy the code
+            assistant.create_mirror_package(_PATH_SRC, _PACKAGE_MAPPING)
     else:
+        assert len(local_pkgs) > 0
+        # PL as a package is distributed together with Lite, so in such case there are more than one candidate
+        package_to_install = "pytorch_lightning" if "pytorch_lightning" in local_pkgs else local_pkgs[0]
+    print(f"Installing package: {package_to_install}")
+
+    # going to install with `setuptools.setup`
+    pkg_path = os.path.join(_PATH_SRC, package_to_install)
+    pkg_setup = os.path.join(pkg_path, "__setup__.py")
+    if not os.path.exists(pkg_setup):
         raise RuntimeError(f"Something's wrong, no package was installed. Package name: {_PACKAGE_NAME}")
+    setup_module = _load_py_module(name=f"{package_to_install}_setup", location=pkg_setup)
+    setup_args = setup_module._setup_args()
+    is_main_pkg = package_to_install == "lightning"
+    print(f"Installing as the main package: {is_main_pkg}")
+    if is_source_install:
+        # we are installing from source, set the correct manifest path
+        with _set_manifest_path(pkg_path, aggregate=is_main_pkg):
+            setuptools.setup(**setup_args)
+    else:
+        setuptools.setup(**setup_args)
+    print("Finished install process.")
```

Comment:
This print is misleading because this code will get called multiple times. I suggest removing it

Reply:
```suggestion
    print("Finished setup configuration.")
```